### DISCUSSION TITLE
Diagnostic messages not showing up in VS or ReSharper

### DISCUSSION THREAD

**Discussion by Euphoric** [xunit/xunit#2405] [Question] [answered]
I'm trying to get IMessageSink into my test fixture. But the messages are not showing up in output.

```
    public class TestServerFixture
    {
        public TestServerFixture(IMessageSink messageSink)
        {
            messageSink.OnMessage(new DiagnosticMessage("======== Test message"));
        }
    }
```

I have xunit.runner.json set up like 
```
{
  "$schema": "https://xunit.net/schema/current/xunit.runner.schema.json",
  "internalDiagnosticMessages": true,
  "diagnosticMessages": true,
  "methodDisplayOptions": "all"
}
```
And I know it is being loaded because the "methodDisplayOptions" takes effect.

It shows up when I run "dotnet test", but it is not shown in either VS test explorer or ReSharper test runner.

I'm using .NET 5, xUnit 2.4.1, xUnit.runner.visualstudio 2.4.3

**Comment by Euphoric** [comment, marked as answer]
Okay. I figures out there is window in ReSharper that shows the output. But I wonder if there is way to make the outputs show in test output.

**Reply by bradwilson** [reply]
I don't think there's much we can do here about where VS or Resharper chooses to show these messages, as we don't control that UI. This should probably become feature requests for either or both of them.
